src/ui/public/utils/aggressive_parse.js

```diff
@@ -0,0 +1,27 @@
+import { isString, startsWith } from 'lodash';
+
+/**
+ * Serializes the given object into a JSON string
+ *
+ * All properties that begin with $ throughout the entire object are omitted.
+ * If a custom JSON serializer function is passed, then the given object is
+ * passed through it before being re-stringified with the native stringify.
+ *
+ * The space argument is passed unaltered to the native stringify.
+ */
+export function toJson(object, jsonFn, space) {
```

Comment:
@spalger The vast majority of the time, we don't need the extra behaviors of angular.toJson, which is why this is an optional argument rather than a direct coupling to angular.toJson.  In the event that you *do* want the extra serializing behaviors of angular.toJson, then you can pass it here.

Ideally we'd just compose our own replacer function with the replacer function that angular uses, but angular doesn't expose it's `toJsonReplacer` function, so we have no way of accessing it.